### PR TITLE
[2.6] Add istio 1.11.7

### PR DESCRIPTION
**Pull request by doflamingo721**
Add support for istio 1.11.7 release:
https://istio.io/latest/news/releases/1.11.x/announcing-1.11.7/

Adds:
Istio v1.11.7
Jaeger Tracing: v1.31.0

Linked issues:
https://github.com/rancher/rancher/issues/36568

mirroring images: 
https://github.com/rancher/image-mirror/pull/212

### CODE REVIEW

File: release.yaml
Comment (aiyengar2) [inline release.yaml:25]: The last released version in `release-v2.6` is `100.1.0+up1.11.4` as seen in https://github.com/rancher/charts/tree/release-v2.6/charts/rancher-istio.

Therefore, in this release we should be releasing `100.1.1+up1.11.7`, based on the versioning scheme outlined in https://github.com/rancher/charts#versioning-charts.

This also means we need to remove [100.1.1+up1.11.4](https://github.com/rancher/charts/tree/dev-v2.6/charts/rancher-istio/100.1.1%2Bup1.11.4) and [100.1.2+up1.11.4](https://github.com/rancher/charts/tree/dev-v2.6/charts/rancher-istio/100.1.2%2Bup1.11.4) from `dev-v2.6`.

We also need to modify the release.yaml accordingly and move it into the commit **before** make charts, since that's how we standardized on making changes (manual changes first, automated changes such as make charts come in the end).

Can you make the appropriate changes in this PR? e.g. 
- Undo the `make charts` commit entirely
- Modify commit 2 to have the istio `package.yaml` to go back to `100.1.1+up1.11.7`
- Merge in the current commit 4 into commit 2 and fix the release.yaml to add `100.1.1+up1.11.7` and remove the other Istio entries (except for 1.10.400, which came from forward porting a chart in release-v2.5)
- run `CHART=rancher-istio VERSION=100.1.1+up1.11.4 make remove && CHART=rancher-istio VERSION=100.1.2+up1.11.4 make remove`. Save this as it's own commit with the message "make remove"
- Rerun `make charts` and save it as the final commit

In the end of the day, there should be exactly 4 commits:
- bump jaeger to 1.31 (packages/rancher-istio/rancher-tracing changes)
- bump istio to 1.11.7 (packages/rancher-istio/rancher-istio changes AND release.yaml changes)
- make remove (charts/, assets/, index.yaml should be modified with removals only)
- make charts (charts/, assets/, index.yaml should be modified with additions only)

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

File: release.yaml
Comment (doflamingo721) [reply release.yaml]: @aiyengar2 Thanks for reviewing. I made the changes as you mentioned above. Can you please review it again and let me know if it looks good?

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
LGTM but `make charts` and `make remove` commits needs to be redone since changes were merged into dev-v2.6 for this chart after you made your changes.

I'd suggest just undoing both (`git reset HEAD~2 --hard`) and rerunning the `make remove` and `make chart` commands in two separate commits.

The final chart produced should **not** undo the changes introduced by the Windows team in https://github.com/rancher/charts/pull/1783.

note: referencing the `make remove` commands in https://github.com/rancher/charts/pull/1765#discussion_r817142734

note: you might be able to just redo the `make charts` commit, but not sure.

**Comment by doflamingo721** [comment]
@aiyengar2 The chart now has the windows changes. Could you please review it again?

**Review by aiyengar2 — COMMENTED** [review]
LGTM! Make sure you merge https://github.com/rancher/istio-installer/pull/49 and tag it to create the image created before merging this PR.

I can approve officially once the other PR is merged and the image is created.

https://hub.docker.com/r/rancher/istio-installer/tags

File: charts/rancher-istio/100.1.1+up1.11.7/Chart.yaml
Comment (aiyengar2) [inline charts/rancher-istio/100.1.1+up1.11.7/Chart.yaml:8]: this annotations should be dropped since we're no longer auto-matching against rancher-kiali-server-crd, which was the one that was providing this GVR in the past. This seems to have been a change introduced in 1.11.7 that was not caught in previous rcs.

Comment (aiyengar2) [reply charts/rancher-istio/100.1.1+up1.11.7/Chart.yaml:8]: e.g. this annotation got dropped https://github.com/rancher/charts/blob/632681382cae5ca209917d06967c19cd28ce9928/charts/rancher-istio/100.0.2%2Bup1.10.4/Chart.yaml#L2

**Comment by doflamingo721** [comment]
@aiyengar2 https://github.com/rancher/istio-installer/pull/49 is merged and tagged. https://hub.docker.com/r/rancher/istio-installer/tags

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
As part of this PR, can you also entirely delete the `packages/rancher-istio/rancher-kiali-server/templates/` directory, which is no longer used since we dropped CRD options (namely https://github.com/rancher/charts/blob/fb4107d95bcefa50963714e8ef124c177eb413ca/packages/rancher-kiali-server/package.yaml#L7, which is why we used this template directory in the first place)?

File: charts/rancher-istio/100.1.1+up1.11.7/Chart.yaml
Comment (doflamingo721) [reply charts/rancher-istio/100.1.1+up1.11.7/Chart.yaml]: @aiyengar2 I was not aware of this annotation. I removed this annotation along with the crd-template directory. Could you please review it again?

**Review by aiyengar2 — APPROVED** [review]

**Pull request merged**
